app/src/main/java/fr/free/nrw/commons/contributions/ContributionViewHolder.java

```diff
@@ -54,13 +58,20 @@
     private final CompositeDisposable compositeDisposable = new CompositeDisposable();
     private final MediaClient mediaClient;
     private boolean isWikipediaButtonDisplayed;
+    private AlertDialog pausingPopUp;
 
     ContributionViewHolder(final View parent, final Callback callback,
         final MediaClient mediaClient) {
         super(parent);
         this.mediaClient = mediaClient;
         ButterKnife.bind(this, parent);
         this.callback = callback;
+
+        // setting the pausingPopUp dialog
```

Comment:
`// Set a dialog indicating that the upload is being paused. This is needed because pausing an upload might take a dozen seconds.`